WireAPI/Sources/WireAPI/Network/PushChannel/PushChannel.swift

```diff
@@ -17,19 +17,141 @@
 //
 
 import Foundation
+import WireUtilitiesPkg
 
-// sourcery: AutoMockable
-/// Make a direct connection to a server to receive update events.
-public protocol PushChannelProtocol {
+final class PushChannel: NSObject, PushChannelProtocol {
 
-    /// Open the push channel and start receiving update events.
-    ///
-    /// - Returns: An async stream of live update event envelopes.
+    private let request: URLRequest
+    private var webSocket: (any WebSocketProtocol)?
+    private var webSocketProvider: (any WebSocketProvider)!
 
-    func open() async throws -> AsyncStream<UpdateEventEnvelope>
+    init(
+        request: URLRequest,
+        minTLSVersion: TLSVersion
+    ) {
+        self.request = request
+        super.init()
+        let factory = URLSessionConfigurationFactory(minTLSVersion: minTLSVersion)
+        let configuration = factory.makeWebSocketSessionConfiguration()
+        webSocketProvider = URLSession(
+            configuration: configuration,
+            delegate: self,
+            delegateQueue: nil
+        )
+    }
 
-    /// Close the push channel and stop receiving update events.
+    init(
+        request: URLRequest,
+        webSocketProvider: any WebSocketProvider
+    ) {
+        self.request = request
+        self.webSocketProvider = webSocketProvider
+    }
 
-    func close()
+    deinit {
+        webSocketProvider.tearDown()
+    }
+
+    func open() async throws -> AsyncThrowingStream<UpdateEventEnvelope, Error> {
+        print("opening new push channel")
```

Comment:
These logs are just print statements now, I want to use the `WireLogger` but it's not accessible yet to `WireAPI`.